frontend/benefit/handler/src/components/applicationReview/salaryBenefitCalculatorView/SalaryBenefitCalculatorView.tsx

```diff
@@ -372,10 +386,181 @@ const SalaryBenefitCalculatorView: React.FC<
+                  </$CalculatorText>
+                </$GridCell>
+              </>
+            )}
+
+          {formik.values.trainingCompensations?.map((item) => (
+            <>
+              <$GridCell
+                $colStart={1}
+                $colSpan={2}
+                style={{ background: 'white' }}
+              >
+                <$CalculatorTableRow
+                  isTotal
+                  style={{ padding: '0 var(--spacing-xs)', height: '100%' }}
+                >
+                  <$ViewField>
+                    {formatStringFloatValue(item.monthlyAmount?.toString())}
+                  </$ViewField>
+                </$CalculatorTableRow>
+              </$GridCell>
+
+              <$GridCell
+                $colStart={3}
+                $colSpan={6}
+                style={{
+                  background: 'white',
+                  marginLeft: '-20px',
+                  paddingLeft: '20px',
+                }}
+              >
+                <$CalculatorTableRow
+                  isTotal
+                  style={{ padding: 0, height: '100%' }}
+                >
+                  <$ViewField>
+                    {t(`${translationsBase}.startEndDates`, {
+                      startDate: convertToUIDateFormat(item.startDate),
+                      endDate: convertToUIDateFormat(item.endDate),
+                      period: diffMonths(
+                        parseDate(item.endDate),
+                        parseDate(item.startDate)
+                      ),
+                    })}
+                  </$ViewField>
+                </$CalculatorTableRow>
+              </$GridCell>
+              <$GridCell $colStart={9} $colSpan={3}>
+                <Button
+                  onClick={() => removeTrainingCompensation(item.id)}
+                  theme="black"
+                  variant="secondary"
+                  iconLeft={<IconMinusCircle />}
+                  style={{ background: 'white' }}
+                >
+                  {t(`${translationsBase}.remove`)}
+                </Button>
+              </$GridCell>
+            </>
+          ))}
+
+          <$GridCell $colStart={1} $colSpan={2}>
+            <$CalculatorText>{fields.monthlyAmount.label}</$CalculatorText>
+          </$GridCell>
+
+          <$GridCell $colStart={3} $colSpan={5}>
+            <$CalculatorText>
+              {t(`${translationsBase}.apprenticeshipPeriodWithDiff`, {
+                period: diffMonths(
+                  parseDate(newTrainingCompensation.endDate),
+                  parseDate(newTrainingCompensation.startDate)
+                ),
+              })}
+            </$CalculatorText>
+          </$GridCell>
+          <$GridCell $colStart={1} $colSpan={1}>
+            <TextInput
+              id={fields.monthlyAmount.name}
+              name={fields.monthlyAmount.name}
+              onChange={(e) =>
+                setNewTrainingCompensation({
```

Comment:
setNewTrainingCompensation((prevValue) => ({...prevValue, monthlyAmount: e.target.value}));` is more safe. Perhaps I would consider also using useReducer here as the state is complex object:
https://reactjs.org/docs/hooks-reference.html#:~:text=useReducer%20is%20usually%20preferable%20to,dispatch%20down%20instead%20of%20callbacks.

Reply:
This is now fixed but not yet using useReducer, imo useReducer is not really necessary here because we only have 1 action type which is to update the newTraining.. object, in addition we use TypeScript so there will be more typing to add